scripts/kconfig/hardenconfig.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from kconfiglib import Symbol, standard_kconfig, standard_config_filename
+
+import csv
+import os
+
+
+def hardenconfig(kconf):
+    kconf.load_config()
+
+    hardened_kconf_filename = os.environ['ZEPHYR_BASE'] + os.sep +  \
+        "scripts" + os.sep + "kconfig" + os.sep + "hardened.csv"
+
+    options = compare_with_hardened_conf(kconf, hardened_kconf_filename, {})
+
+    display_results(options)
+
+
+class Option(object):
+
+    def __init__(self, name: str, recommended: str, current=None, symbol=None):
+        self.name = name
+        self.recommended = recommended
+        self.current = current
+        self.symbol = symbol
+
+        if current is 'NA':
+            self.result = 'NA'
+        elif recommended == current:
+            self.result = 'PASS'
+        else:
+            self.result = 'FAIL'
+
+
+def compare_with_hardened_conf(kconf, hardened_kconf_filename,  options):
+    options = {}
+
+    with open(hardened_kconf_filename) as csvfile:
+        csvreader = csv.reader(csvfile)
+        for row in csvreader:
+            name = row[0]
+            recommended = row[1]
+            try:
+                symbol = kconf.syms[name]
+                current = symbol.str_value
+            except KeyError:
+                symbol = None
+                current = 'NA'
```

Comment:
A better "not applicable" value might be `current = None`.

The idiomatic way to test for it later is `if current is None:`.